source/agora/test/Fee.d

```diff
@@ -219,3 +219,39 @@ unittest
     auto bad_node = nodes[0];
     assert(bad_node.getBlockHeight() == Height(0));
 }
+
+// Fees from the previous block should not trigger a new block creation
+// where there is no TXs in the pool
+unittest
+{
+    import std.exception;
+    import core.exception : AssertError;
+
+    TestConf conf = {
+        txs_to_nominate : 1,
+        payout_period : 1,
+        quorum_threshold : 100
+    };
+    auto network = makeTestNetwork(conf);
+    network.start();
+    scope(exit) network.shutdown();
+    scope(failure) network.printLogs();
+    network.waitForDiscovery();
+
+    auto nodes = network.clients;
+    auto node_1 = nodes[0];
+
+    // Get the genesis block, make sure it's the only block externalized
+    auto blocks = node_1.getBlocksFrom(0, 2);
+    assert(blocks.length == 1);
+
+    Transaction[] txs;
+    // create enough tx's for a single block
+    txs = blocks[$ - 1].spendable().map!(txb => txb
+        .deduct(Amount.UnitPerCoin).sign()).array();
+    // Send a single TX with fees to a node
+    node_1.putTransaction(txs[0]);
+    network.expectBlock(Height(1), blocks[0].header);
+    // The fees from the last block should not trigger a new block creation
+    assertThrown!AssertError(network.expectBlock(Height(2), blocks[0].header));
```

Comment:
We should *never* test for `AssertError`. The way to do it would be to instrument a validator.